exporter/stats/stackdriver/stackdriver_test.go

```diff
@@ -525,6 +527,70 @@ func TestEqualAggWindowTagKeys(t *testing.T) {
 	}
 }
 
+func TestExporter_createMeasure(t *testing.T) {
+	oldGetMetricDescriptor := getMetricDescriptor
+	oldCreateMetricDescriptor := createMetricDescriptor
+
+	defer func() {
+		getMetricDescriptor = oldGetMetricDescriptor
+		createMetricDescriptor = oldCreateMetricDescriptor
+	}()
+
+	key, _ := tag.NewKey("test-key-one")
+	m, err := stats.NewMeasureFloat64("test-measure", "measure desc", "unit")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer stats.DeleteMeasure(m)
+
+	view, err := stats.NewView("cumview", "desc", []tag.Key{key}, m, stats.CountAggregation{}, stats.Cumulative{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	data := stats.CountData(0)
+	vd := newTestCumViewData(view, time.Now(), time.Now(), &data, &data)
+
+	e := &Exporter{
+		createdViews: make(map[string]*metricpb.MetricDescriptor),
+	}
+
+	var getCalls, createCalls int
+	getMetricDescriptor = func(ctx context.Context, c *monitoring.MetricClient, mdr *monitoringpb.GetMetricDescriptorRequest) (*metric.MetricDescriptor, error) {
+		getCalls++
+		return nil, status.Error(codes.NotFound, "")
+	}
+	createMetricDescriptor = func(ctx context.Context, c *monitoring.MetricClient, mdr *monitoringpb.CreateMetricDescriptorRequest) (*metric.MetricDescriptor, error) {
+		createCalls++
+		return &metric.MetricDescriptor{
+			DisplayName: "display",
+			Description: "desc",
+			Unit:        "unit",
+			Type:        "hello",
+			MetricKind:  metricpb.MetricDescriptor_CUMULATIVE,
+			ValueType:   metricpb.MetricDescriptor_INT64,
+			Labels:      newLabelDescriptors(vd.View.TagKeys()),
+		}, nil
+	}
+
+	ctx := context.Background()
+	if err := e.createMeasure(ctx, vd); err != nil {
+		t.Errorf("Exporter.createMeasure() error = %v", err)
+	}
+	if err := e.createMeasure(ctx, vd); err != nil {
+		t.Errorf("Exporter.createMeasure() error = %v", err)
+	}
+	if count := getCalls; count != 1 {
```

Comment:
s/count != 1/count != 2?

Reply:
No, 1. Because we will cache the MetricDescriptor in the createdViews map. We don't do an RPC each time we call createMeasure.

Reply:
I see